flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/optimize/StreamCommonSubGraphBasedOptimizer.scala

```diff
@@ -83,25 +87,42 @@ class StreamCommonSubGraphBasedOptimizer(planner: StreamPlanner)
         isSinkBlock = true)
       block.setOptimizedPlan(optimizedTree)
       return sinkBlocks
+    } else {
```

Comment:
nit: It seems that reverting this part of the changes won't be an issue. How about reverting it?